azurerm/data_source_management_group.go

```diff
@@ -0,0 +1,109 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/resources/mgmt/2018-03-01-preview/management"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmManagementGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmManagementGroupRead,
+
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"display_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"parent_management_group_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"subscription_ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+		},
+	}
+}
+
+func dataSourceArmManagementGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).managementGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	groupId := d.Get("group_id").(string)
+
+	recurse := true
+	resp, err := client.Get(ctx, groupId, "children", &recurse, "", managementGroupCacheControl)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Management Group %q doesn't exist - removing from state", d.Id())
```

Comment:
Not a huge deal, but I thought we used to log these as `WARN` in some other providers I worked with? 🤔

Reply:
actually this should be a failure since this is a data source 🤦‍♂️